services/docker/docker-cloud-automated.tester.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const Joi = require('joi')
+const t = (module.exports = require('../tester').createServiceTester())
+const { dockerBlue } = require('./docker-helpers')
+
+const isAutomatedBuildStatus = Joi.string().valid('automated', 'manual')
+
+t.create('docker cloud automated build (valid, user)')
+  .get('/jrottenberg/ffmpeg.json')
+  .expectBadge({
+    label: 'docker build',
+    message: isAutomatedBuildStatus,
+  })
+
+t.create('docker cloud automated build (not found)')
+  .get('/badges/not-a-real-repo.json')
+  .intercept(nock =>
+    nock('https://cloud.docker.com/')
+      .get('/api/build/v1/source/?image=badges/not-a-real-repo')
```

Comment:
I think nock is probably barking on `...source/?image` vs `...source?image...` you may also want to use a template string and do a `${encodeURIComponent(...)}` on that part if it keeps failing